tools/flaky-test-retryer/github_commenter.go

```diff
@@ -21,19 +21,175 @@ limitations under the License.
+
 	"github.com/knative/test-infra/shared/ghutil"
+	"github.com/google/go-github/github"
+)
+
+const (
+	maxRetries            = 3
+	maxFailedTestsToPrint = 8
+)
+
+var (
+	identifier      = "<!--AUTOMATED-FLAKY-RETRYER-->"
+	commentTemplate = "%s\nThe following tests are currently flaky. Running them again to verify..." +
+		"\n\nTest name | Retries\n--- | ---\n%s\n\n%s"
 )
 
 // GithubClient wraps the ghutil Github client
 type GithubClient struct {
-	*ghutil.GithubClient
+	ghutil.GithubOperations
+	Login  string
+	Dryrun bool
 }
 
 // NewGithubClient builds us a GitHub client based on the token file passed in
-func NewGithubClient(githubAccount string) (*GithubClient, error) {
+func NewGithubClient(githubAccount string, dryrun bool) (*GithubClient, error) {
 	ghc, err := ghutil.NewGithubClient(githubAccount)
 	if err != nil {
 		return nil, err
 	}
-	return &GithubClient{ghc}, err
+	user, err := ghc.GetGithubUser()
+	if err != nil {
+		return nil, err
+	}
+	return &GithubClient{ghc, *user.Login, dryrun}, nil
+}
+
+// PostComment posts a new comment on the PR specified in JobData, retrying the job that triggered it.
+// The comment body is dynamically built based on previous retry comments on this PR, and any old
+// comments are removed before the new one is posted.
+func (gc *GithubClient) PostComment(jd *JobData, outliers []string) error {
+	oldComment, err := gc.getOldComment(jd.Refs[0].Org, jd.Refs[0].Repo, jd.Refs[0].Pulls[0].Number)
+	if err != nil {
+		return err
+	}
+	oldEntries, err := parseEntries(oldComment)
+	if err != nil {
+		return err
+	}
+	if _, ok := oldEntries[jd.JobName]; !ok {
+		oldEntries[jd.JobName] = 0
+	}
+	newComment, canRetry := buildNewComment(jd, oldEntries, outliers)
+	if !canRetry {
+		return fmt.Errorf("expended all %d retries", maxRetries)
+	}
+	if gc.Dryrun {
+		logWithPrefix(jd, "[dry run] Comment not updated. See it here:\n%s\n", newComment)
+		return nil
+	}
+	if oldComment != nil {
+		if err := gc.DeleteComment(jd.Refs[0].Org, jd.Refs[0].Repo, oldComment.GetID()); err != nil {
+			return err
+		}
+	}
+	_, err = gc.CreateComment(jd.Refs[0].Org, jd.Refs[0].Repo, jd.Refs[0].Pulls[0].Number, newComment)
+	return err
+}
+
+// getOldComment queries the GitHub PR specified and gets the comment made by us. If no comment
+// is found, we do not error, since we will be creating a new one anyways.
+func (gc *GithubClient) getOldComment(org, repo string, pull int) (*github.IssueComment, error) {
+	comments, err := gc.ListComments(org, repo, pull)
+	if err != nil {
+		return nil, err
+	}
+	// this robot should only leave one comment in a given PR. Collect everything with our identifier,
+	// just in case.
+	var matches []*github.IssueComment
+	for _, comment := range comments {
+		match, err := regexp.Match(identifier, []byte(comment.GetBody()))
+		if err != nil {
+			return nil, err
+		}
+		if match && *comment.GetUser().Login == gc.Login {
+			matches = append(matches, comment)
+		}
+	}
+
+	switch len(matches) {
```

Comment:
I dont think this switch is necessary. Have just one var and return var, err

Reply:
done